Software/Development/SemesterProject/App/analogSignal.c

```diff
@@ -0,0 +1,98 @@
+//###########################################################################
+//FILE: signal.c
+//
+//Project: Electric Go-kart
+//
+//Author: Nicolai, 6/3/19
+//
+//Notes:    Signal. This will create the signals for each ADC.
+//
+//
+//###########################################################################
+
+//###########################################################################
+
+//
+// Includes
+//
+
+#include "DSP28x_Project.h"     // Device Headerfile and Examples Include File
+#include "AnalogSignal.h"
+#include <math.h>
+
+
+//
+// Quasi-global variables definition
+//
+#define ANALOG_EXECUTION_FREQ 20000
+
+/*
+ * The analog signal constructor initialize all defined variables to the selected analog signal struct.
+ * Furthermore it calls the method filterParameters() to calculate the necessary parameters for the filter.
+ */
+void Signal_Constructor(AnalogSignal *analogSignal, char filterType, int filterOrder,
+                       int cutoffFreq, Uint16 adcChannel, int threshold[2])
+{
+    analogSignal->filterType = filterType;
+    analogSignal->filterOrder = filterOrder;
+    analogSignal->cutoffFreq = cutoffFreq;
+
+    analogSignal->adcChannel = adcChannel;
+
+    analogSignal->threshold[0] = threshold[0];
+    analogSignal->threshold[1] = threshold[1];
+
+    analogSignal->oldValue = 0;
+
+    calculateFilterParameters(analogSignal, filterType, filterOrder, cutoffFreq);
+}
+
+/*
+ * The filter parameters are calculated from the specified filter type, order and cut-off frequency.
+ * The B parameter is calculated by using the execution frequency, so this should be confirmed with a clever person.
+ *
+ * The parameters A and B are calculated as in the previous project (Ask Nicolas for the reference)
+ * For now its only able to calculated the parameters for a first order low-pass filter, if it is found
+ * necessary to use a higher order filter this should be implemented.
+ */
+void calculateFilterParameters(AnalogSignal *analogSignal, char filterType, int filterOrder, int cutoffFreq)
+{
+    float parameterA, pi = 3.14;

```

Comment:
Why not writing more decimals for pi? 💃

Reply:
Why write more decimals for pi?

Reply:
Because it doesn't take more resources to write down more decimals! 👯

Reply:
I have used the file from Lajos now